src/Common/src/Interop/Linux/libc/Interop.inotify.cs

```diff
@@ -14,8 +14,30 @@ internal static partial class libc
         [DllImport(Libraries.Libc, SetLastError = true)]
         internal static extern int inotify_add_watch(int fd, string pathname, uint mask);
 
-        [DllImport(Libraries.Libc, SetLastError = true)]
-        internal static extern int inotify_rm_watch(int fd, int wd);
+        [DllImport(Libraries.Libc, SetLastError = true, EntryPoint = "inotify_rm_watch")]
+        private static extern int inotify_rm_watch_extern(int fd, int wd);
+
+        internal static int inotify_rm_watch(int fd, int wd)
+        {
+            int result = Interop.libc.inotify_rm_watch_extern(fd, wd);
+            if (result < 0)
+            {
+                int hr = System.Runtime.InteropServices.Marshal.GetLastWin32Error();
+                if (hr == Interop.Errors.EINVAL)
+                {
+                    // This specific case means that there was a deleted event in the queue that was not processed
+                    // so this call is expected to fail since the WatchDescriptor is not longer valid and was cleaned
+                    // up automatically by the OS.
+                    result = 0;
+                }
+                else
+                {
+                    System.Diagnostics.Debug.Assert(false, "inotify_rm_watch failed with " + hr);
```

Comment:
I optimized for what the current code has, which is two calling sites checking the same thing. I did this because it's more likely that the current code will have churn and fix bugs and need the same checks; I figure that it's more unlikely that we will add future calls to inotify_rm_watch, let alone calls that should not take this check into consideration since it could always return this failure code. 

Thoughts?